doc/sphinx/proof-engine/ssreflect-proof-language.rst

```diff
@@ -4086,8 +4197,19 @@ will generally fail to perform congruence simplification, even on
 rather simple cases. We therefore provide a more robust alternative in
 which the function is supplied:
 
-.. tacn:: congr {? @natural } @term
-   :name: congr
+.. tacn:: congr {? @natural } @one_term {? @ssrdgens }
+   :name: congr (ssreflect)
+
+   .. insertprodn ssrdgens ssrdgens_tl
+
+   .. prodn::
+      ssrdgens ::= : @ssrgen {? @ssrdgens_tl }
+      ssrgen ::= {? @ssrdocc } @cpattern
+      ssrdgens_tl ::= %{ {+ @ident } %} @cpattern {? @ssrdgens_tl }
+      | %{ {+ @ident } %}
+      | %{ @clear_switch %} @cpattern {? @ssrdgens_tl }
+      | / {? @ssrdgens_tl }
+      | @cpattern {? @ssrdgens_tl }
```

Comment:
Here the grammar is too messy/permissive. Let's try this.
```
ssrdgens  := : (ssrgen | ssrgen / ssrgen)    (* only one / possible *)
ssrgen := cpattern (clear_switch | cpattern)*
clear_switch := { ident+ }
```
Examples:
```
: x
: x {y} z
: {3 4 5}(t _)
: _ / _
```
Non examples:
```
: {H}
: {H} / {H}
: {H} / x
```

Reply:
The things you want to hide are not supported?

`clear_switch` is already defined like this elsewhere.  So I inlined your `{ ident }+` above.

Check what I did.